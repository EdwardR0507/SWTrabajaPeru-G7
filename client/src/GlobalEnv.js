module.exports = {
  //host: process.env.HOST || "http://localhost:4000",
  host: "https://ws-trabajaperu.herokuapp.com",
};
