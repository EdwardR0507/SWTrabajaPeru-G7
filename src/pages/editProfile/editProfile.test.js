/*Importamos las librerias principales*/
import React from 'react';
import { shallow } from 'enzyme';
import { rest } from 'msw'
import { setupServer } from 'msw/node';
import { render, screen,cleanup, fireEvent } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router, Route } from 'react-router-dom';
import GlobalEnv from '../../GlobalEnv';
import EditProfile from './index';
import { Snackbar } from '@material-ui/core';
/*Renderizado de la vista EditProfile*/
describe("<EditProfile></EditProfile>", () => {
  const server = setupServer(
    rest.post(`${GlobalEnv.host}/user-auth`, (req, res, ctx) => {
      return res(ctx.json({ greeting: 'hello there' }))
    }),
  )

  beforeAll(() => { server.listen()})
  afterEach(() => { server.resetHandlers() })
  afterAll(() => { server.close() })

  it('render "EditProfile"', () => {
    const history = createMemoryHistory()
    const wrapper = shallow(<Router history={history}>
      <EditProfile />
    </Router>);
    expect(wrapper).toMatchSnapshot()
  });
  it('show Cargando in editprofile', () => {
    const history = createMemoryHistory();
    server.use(
      rest.post(`${GlobalEnv.host}/user-auth`, (req, res, ctx) => {
        return res(ctx.status(500))
      }),
    )
    render(<Router history={history}>
      <EditProfile /> 
  </Router>);
  expect(screen.getByRole('edit-profile')).toHaveTextContent('Cargando...')
  })
  it('show 500', () => {
    const history = createMemoryHistory();
    server.use(
      rest.post(`${GlobalEnv.host}/user-auth`, (req, res, ctx) => {
        return res(ctx.status(500))
      }),
    )
    const result = render(<Router history={history}>
      <EditProfile /> 
  </Router>);
  fireEvent(<Snackbar />, 'onClose');
  expect(result.current.open).toBe(false)
  })
});