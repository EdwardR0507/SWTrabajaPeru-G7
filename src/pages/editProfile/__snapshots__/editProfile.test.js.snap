// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditProfile></EditProfile> render "EditProfile" 1`] = `ShallowWrapper {}`;
